pkg/server/api/agent/v1/service.go

```diff
@@ -283,21 +283,26 @@ func (s *Service) AttestAgent(stream agentv1.Agent_AttestAgentServer) error {
 		}
 	}
 
-	agentID := attestResult.AgentID
-	log = log.WithField(telemetry.AgentID, agentID)
-	rpccontext.AddRPCAuditFields(ctx, logrus.Fields{telemetry.AgentID: attestResult.AgentID})
-
-	if err := idutil.CheckAgentIDStringNormalization(agentID); err != nil {
-		return api.MakeErr(log, codes.Internal, "agent ID is malformed", err)
-	}
-
-	agentSpiffeID, err := spiffeid.FromString(agentID)
+	// Ideally we'd do stronger validation that the ID is within the Node
+	// Attestors scoped area of the reserved agent namespace, but historically
+	// we haven't been strict here and there are deployments that are emitting
+	// such IDs.
+	// TODO: Enforce that IDs produced by Node Attestors are in the reserved
```

Comment:
Is there a version where we'll add this kind of restrictions? if it is something we plan to do soon, may we can add a warning to start noticing users about it will no longer valid in a near feature?

Reply:
Yes, good call. I'll add a warning.